packages/components/src/internal/components/editable/utils.test.ts

```diff
@@ -637,6 +562,216 @@ describe('getUpdatedDataFromGrid', () => {
         expect(updatedData).toHaveLength(1);
     });
 
+    test('edited row date and time value changes', () => {
+        let updatedData = getUpdatedDataFromGrid(
+            fromJS({
+                448: {
+                    RowId: 448,
+                    Date: '2020-12-23 14:34',
+                    Time: '01:10:00',
+                },
+            }),
+            [
+                Map<string, any>({
+                    RowId: 448,
+                    Date: '2020-12-23 14:35',
+                    Time: '01:10:00',
+                }),
+            ],
+            'RowId',
+            queryInfo
+        );
+        expect(updatedData[0]).toStrictEqual({
+            Date: '2020-12-23 14:35',
+            RowId: 448,
+        });
+
+        updatedData = getUpdatedDataFromGrid(
+            fromJS({
+                448: {
+                    RowId: 448,
+                    Date: '2020-12-23 14:34',
+                    Time: '01:10:00',
+                },
+            }),
+            [
+                Map<string, any>({
+                    RowId: 448,
+                    Date: '2020-12-24 14:34',
+                    Time: '01:10:00',
+                }),
+            ],
+            'RowId',
+            queryInfo
+        );
+        expect(updatedData[0]).toStrictEqual({
+            Date: '2020-12-24 14:34',
+            RowId: 448,
+        });
+
+        updatedData = getUpdatedDataFromGrid(
+            fromJS({
+                448: {
+                    RowId: 448,
+                    Date: '2020-12-23 14:34',
+                    Time: '01:10:00',
+                },
+            }),
+            [
+                Map<string, any>({
+                    RowId: 448,
+                    Date: '2020-12-INVALID 14:34',
+                    Time: '01:10:00',
+                }),
+            ],
+            'RowId',
+            queryInfo
+        );
+        expect(updatedData[0]).toStrictEqual({
+            Date: '2020-12-INVALID 14:34',
+            RowId: 448,
+        });
+
+        updatedData = getUpdatedDataFromGrid(
+            fromJS({
+                448: {
+                    RowId: 448,
+                    Date: '2020-12-23 14:34',
+                    Time: '01:10:00',
+                },
+            }),
+            [
+                Map<string, any>({
+                    RowId: 448,
+                    Date: new Date('2020-12-23 14:34'),
+                    Time: '01:10:00',
+                }),
+            ],
+            'RowId',
+            queryInfo
+        );
+        expect(updatedData[0]).toStrictEqual({
+            Date: new Date('2020-12-23 14:34'),
+            RowId: 448,
+        });
```

Comment:
Also might be useful to include a test for also updating the time field.

Reply:
good catch. done